borg/repository.py

```diff
@@ -591,6 +602,28 @@ def get_fd(self, segment):
             self.fds[segment] = fd
             return fd
 
+    def _sync_and_close(self, fd):
+        fd.flush()
+        os.fdatasync(fd.fileno())
+        fd.close()
+
+    def close_segment(self, sync=False):
+        if len(self._write_fds) == self.MAX_PENDING_WRITE_FDS:
+            self._sync_and_close(self._write_fds.popleft())
```

Comment:
also, shouldn't this whole if-block be moved to the line directly above self._write_fds.append()?